src/app/core/admin/admin-entity-details/admin-entity-panel-component/admin-entity-panel-component.component.ts

```diff
@@ -163,4 +200,39 @@ export class AdminEntityPanelComponentComponent implements OnInit {
       };
     }
   }
+
+  /**
+   * Opens a dialog showing the related entity's column selection and ordering.
+   * When "Apply" is clicked, updates the columns in the config.
+   * Changes are saved when the main form's Save button is clicked (config is part of the form).
+   */
+  openRelatedEntityDetailsConfig(): void {
+    if (!this.entityConstructor) {
+      return;
+    }
+
+    const dialogRef = this.dialog.open(AdminRelatedEntityDetailsComponent);
+
+    dialogRef.componentInstance.entityConstructor = this.entityConstructor;
+
+    // Pass the current columns to show as active fields in the dialog
+    dialogRef.componentInstance.currentColumns = this.activeFields.map((col) =>
+      typeof col === "string" ? col : col.id,
+    );
+
+    dialogRef.afterClosed().subscribe((updatedFieldIds: string[]) => {
+      // Only update if user clicked Apply (updatedFieldIds is defined)
+      // undefined means Cancel was clicked
+      if (updatedFieldIds !== undefined) {
+        // Update the columns configuration with the new field order
+        // This will be saved when the main form's Save button is clicked
+        this.config.config = {
+          ...this.config.config,
+          columns: updatedFieldIds,
+        };
+
+        this.activeFields = updatedFieldIds;
```

Comment:
your approach looks good to me in general. Seems smart to overwrite the fields available with the new returned from the dialog.

But probably this should include any fields that related entity has now, not only those that the user pulled into the actively selected/visible fields in the dialog.

Do updates work now?